frame/session/src/lib.rs

```diff
@@ -358,18 +386,10 @@ pub trait WeightInfo {
 	fn purge_keys() -> Weight;
 }
 
-pub trait Trait: frame_system::Trait {
+pub trait Trait: ValidatorIdentification<<Self as frame_system::Trait>::AccountId> + frame_system::Trait {
```

Comment:
Not a huge fan of this change. The `FullIndentification` is only needed for `historical` sessions, that's why it was part of that other trait. What's the reasoning behind this change?

Reply:
hmm, session uses a bunch of `T::ValidatorId` which is in `ValidatorIdentification` trait. Maybe split `ValidatorIndentification` into two traits `ValidatorIdentification` and `FullValidatorIdentification` then the normal session won't be related to `FullIdentification`?

```rust
pub trait ValidatorIdentification<AccountId> {
	/// A stable ID for a validator.
	type ValidatorId: Member + Parameter;

	/// A conversion from account ID to validator ID.
	///
	/// Its cost must be at most one storage read.
	type ValidatorIdOf: Convert<AccountId, Option<Self::ValidatorId>>;
}

pub trait FullValidatorIdentification<AccountId>: ValidatorIdentification<AccountId> {
	/// Full identification of the validator.
	type FullIdentification: Parameter;

	/// A conversion from validator ID to full identification.
	///
	/// This should contain any references to economic actors associated with the
	/// validator, since they may be outdated by the time this is queried from a
	/// historical trie.
	///
	/// It must return the identification for the current session index.
	type FullIdentificationOf: Convert<Self::ValidatorId, Option<Self::FullIdentification>>;
}
```

Reply:
@tomusdrw I splitted out `FullValidatorIdentification` trait for the historical sessions in https://github.com/paritytech/substrate/pull/7127/commits/004bde7857ec5881dc6b1156069d86f513c13add, let me know if you still are unhappy with that change.